src/gpio.rs

```diff
@@ -269,6 +400,32 @@ macro_rules! impl_base {
 
                 Ok(())
             }
+
+            #[cfg(all(not(feature = "riscv-ulp-hal"), feature = "alloc"))]
+            fn enable_interrupt(&mut self) -> Result<(), EspError> {
+                esp!(unsafe { gpio_intr_enable(self.pin()) })?;
+
+                Ok(())
+            }
+
+            #[cfg(all(not(feature = "riscv-ulp-hal"), feature = "alloc"))]
+            fn disable_interrupt(&mut self) -> Result<(), EspError> {
+                esp!(unsafe { gpio_intr_disable(self.pin()) })?;
+                esp!(unsafe { gpio_set_intr_type(self.pin(), gpio_int_type_t_GPIO_INTR_DISABLE) })?;
+                unsafe { unregister_irq_handler(self.pin() as usize) };
```

Comment:
Not sure about this. Would normally expect functions called `enable_interrupt` and `disable_interrupt` to be symmetrical. But having `disable_interrupt` also unregister the interrupt handler might be a little bit unexpected. Then again, I think these are private functions, so may not matter. But maybe rename to `unregister_interrupt` to carlify what it does?

Reply:
I created this private method to gather all call required to disable the interrupt on the pin and clean things up. It's called from the `reset` method.